src/frontend/src/scheduler/local.rs

```diff
@@ -84,17 +93,46 @@ impl LocalQueryExecution {
     /// will be embedded in exchange source, so we can always convert a query into a plan fragment.
     ///
     /// We remark that the boundary to determine which part should be executed on the frontend and
-    /// which part should be executed on the backend is the first exchange operator when looking
+    /// which part should be executed on the backend is `the first exchange operator` when looking
     /// from the the root of the plan to the leaves. The first exchange operator contains
     /// the pushed-down plan fragment.
+    ///
+    /// We also remark that there are at most two stages during local execution mode, meaning that
+    /// `the first exchange operator` is the only possible exchange operator in the whole plan.
+    /// And it is possible that there is no exchange operator at all, e.g. `select 1+2;`
     fn create_plan_fragment(&self) -> Result<PlanFragment> {
-        let stage = self
-            .query
-            .stage_graph
-            .stages
-            .get(&self.query.root_stage_id())
-            .unwrap();
-        let plan_node_prost = self.convert_plan_node(&*stage.root)?;
+        let root_stage_id = self.query.root_stage_id();
+        let root_stage = self.query.stage_graph.stages.get(&root_stage_id).unwrap();
+        let second_stage_id = self.query.stage_graph.get_child_stages(&root_stage_id);
+        let plan_node_prost = match second_stage_id {
+            None => {
+                debug!("Local execution mode converts a plan with a single stage");
+                self.convert_plan_node(&*root_stage.root, None)?
+            }
+            Some(second_stage_id) => {
+                debug!("Local execution mode converts a plan with two stages");
+                if second_stage_id.is_empty() {
+                    // This branch is defensive programming. The semantics should be the same as
+                    // `None`.
+                    self.convert_plan_node(&*root_stage.root, None)?
+                } else {
+                    // We should only have one child stage of the root stage for now.
+                    assert_eq!(second_stage_id.len(), 1);
+                    let second_stage_id = second_stage_id.iter().next().unwrap();
```

Comment:
The child stages of root maybe more than 1, think about following sql:
`select * from t1, t2 where t1.a = t2.a`, in this case we have plan like 

```
                   HashJoin
               /                     \
          Exchange      Exchange
              /                        \
         TableScan         TableScan
```
You can refer to https://github.com/singularity-data/risingwave/blob/408e9fb5249b12b1b457287adc4deba13c301f18/src/frontend/src/scheduler/distributed/stage.rs#L354 for child plan fragment and exchange id mapping.

Reply:
I thought `select * from t1, t2 where t1.a = t2.a` is not considered as a `point query` and thus is executed in the `normal distributed mode` instead of `local execution mode`. 🤔

Reply:
Choosing the appropriate plan should be the optimizer's responsibility, and the scheduler should not have such strong assumptions about the plan. The `local execution mode` is optimized for `point query`, but it doesn't mean the user can't execute hash join/sort-merge join. 
The example `SELECT pk, t1.a, t1.fk, t2.b FROM t1, t2 WHERE t1.fk = t2.pk AND t1.pk = 114514
` picks lookup join only when t2.pk has index or is a primary key on it, and chooses hash join when it doesn't have an index.